fgl.cabal

```diff
@@ -69,17 +64,14 @@ library {
                     , transformers
                     , array
 
-    if flag(containers042)
-        build-depends:    containers >= 0.4.2
-                        , deepseq >= 1.1.0.0 && < 1.5
-    else
-        build-depends:    containers < 0.4.2
+    build-depends:    containers >= 0.5.0
+                    , deepseq >= 1.1.0.0 && < 1.5
 
     if impl(ghc >= 7.2) && impl(ghc < 7.6)
         build-depends:
             ghc-prim
 
-    ghc-options:      -Wall
+    ghc-options:      -Wall -O2
```

Comment:
I don't know if it makes a difference over `-O`. We should probably benchmark that. No, Hackage does not prevent packages from specifying an optimization level (`containers` does so, and `vector`, and all sorts of others).